### PR TITLE
[BUGFIX] Fix missing paths for Icons

### PR DESCRIPTION
**Pull request by jmverges**
Close: #253

### CODE REVIEW

File: Classes/Service/ConfigurationService.php [hunk 1]
```diff
@@ -351,4 +365,16 @@ protected function sendDisabledContentWarning($templatePathAndFilename) {
...
+	/**
+	 * @param string $sourceFile
+	 * @param integer $maxWidth
+	 * @param integer $maxHeight
+	 * @return string
+	 */
+	protected function resizeImage($sourceFile, $maxWidth, $maxHeight) {
+		$cmd = GeneralUtility::imageMagickCommand('convert', "+profile '*' -geometry " . $maxWidth . 'x' . $maxHeight . "!   '" . $sourceFile . "' '/var/www/typo3temp/pics/" . sha1($sourceFile) . ".png'");
+		CommandUtility::exec($cmd);
+		return '/typo3temp/pics/' . sha1($sourceFile) . '.png';
```
Comment (xf-) [inline Classes/Service/ConfigurationService.php]: Please use a utility and maybe the upload folder. temp is often cleared and wizard is cached

File: Classes/Service/ConfigurationService.php [hunk 2]
```diff
@@ -312,7 +313,20 @@ protected function buildAllWizardTabsPageTsConfig($wizardTabs) {
...
+		if (0 === strpos($icon, '../')) {
+			$icon = substr($icon, 2);
+		}
+
+		if ('/' === $icon[0]) {
+			$icon = realpath(PATH_site . $icon);
+		}
+
+		if (TRUE === file_exists($icon)) {
+			$icon = $this->resizeImage($icon, 24, 24);
```
Comment (xf-) [inline Classes/Service/ConfigurationService.php]: I would hard code it. or if possible read a t3 setting

**Comment by xf-** [comment]
depends on https://github.com/FluidTYPO3/flux/pull/865

File: Classes/Service/ConfigurationService.php [hunk 3]
```diff
@@ -34,6 +36,16 @@
 class ConfigurationService extends FluxService implements SingletonInterface {
 
 	/**
+	 * Default Width for icon
+	 */
+	const DEFAULT_WIDTH = 24;
```
Comment (xf-) [inline Classes/Service/ConfigurationService.php]: ICON_WIDTH - never know what will be added

**Comment by xf-** [comment]
+1 - works in 6.3 and 7-dev

**Comment by julianxhokaxhiu** [comment]
I'm on version 4.3.1 and I have this patch inside, but actually I'm having anyway the issue of missing icons. The paths are absolute. Is there a typoscript settings or something to be set? Also reading the code patch why you [replaced the var here](https://github.com/jmverges/fluidcontent/commit/325125ad6e518837dd21a00f44d3af4ca096dc62#diff-f9a5b52402d3ab46c235565581665470L329)? Was it supposed to be right?

**Comment by xf-** [comment]
You need also flux dev version for the wizard. We added a ImageUtility central in flux (because it could be interesting for other functions as well). The version isn't released yet.

**Comment by julianxhokaxhiu** [comment]
Gotcha, I'm gonna fix this on monday then :)

**Pull request merged**